### ISSUE TITLE
Insurance for Dutch Parcels: difference between SDK and API Documentation

### ISSUE THREAD

**Issue by LeonidasJP** [myparcelnl/sdk#5]
Hi Reindert,

According to the API Documentation, [chapter 6.A.4](https://myparcelnl.github.io/api/#6_A_4), example 3 "NL Insured", the insured amount should be injected in cents, and be either 5000, 25000, 50000 or 500000.

But when I view the code in the SDK, Model\MyParcelConsignment::setInsurance() accepts the Insured amount in whole euros, and the function also accepts values between 1000 and 4500 euros, in steps of 500. 

These values are not enforced when setInsurance gets called; therefore I'd still be able to inject invalid values like "1Aje83", which would eventually lead to an insurance for just € 1,-

Would it be an option to enhance setInsurance to also check for a valid Insurance value, and add a note in the SDK-documentation that the Insurance value should be provided in whole euros?
Perhaps the API documentation should be updated as well, if the 1000-4500 range should be supported...

**Comment by reindert-vetter** [comment]
I thought I had resolved this before. Could you check this?

**Comment by LeonidasJP** [comment]
Currently, in the Master-branch, SDK\Model\MyParcelConsignment::setInsurance() is formed like this:

```php
    /**
     * Insurance price for the package.
     *
     * Composite type containing integer and currency. The amount is without decimal
     * separators.
     * Pattern: [50, 250, 500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000]
     * Required: No
     *
     * @param int $insurance
     *
     * @return $this
     */
    public function setInsurance($insurance)
    {
        if (!$this->canHaveOption()) {
            $insurance = 0;
        }
        $this->insurance = $insurance;
        return $this;
    }
```

There is no check if $insurance is even an integer, or the inserted integer is allowed... So it does not seem to be fixed in the current release...

**Comment by reindert-vetter** [comment]
This is solved in version [v1.2.0-beta.2](https://github.com/myparcelnl/sdk/releases/tag/v1.2.0-beta.2). Could you to check if its working properly?
